### PR TITLE
77-Chore: Tests for repeated merkle drops

### LINKED ISSUE #77: Chore: New tests for repeated merkle drop
### User Story
**As** prime DAO, **I want** to allocation v2 staggered via the MerkleDrop contract.

### Status quo
- currently there are no specs testing multiple drops through the merkle contract

### AC
- there are test scenarios that test multiple merkle drops

### PR DESCRIPTION
**Pull request by fabianschu**
closes #77

### CODE REVIEW

**Review by Zitzak — DISMISSED** [review]

File: test/unit/MerkleDrop.spec.js [hunk 1]
```diff
@@ -71,9 +71,9 @@ const setupInitialState = async (contractInstances, initialState) => {
   const {
     withProof,
     trancheExpired,
-    forwardBlocks,
     zeroAllocation,
     incorrectProof,
+    withoutSeededAllocation,
   } = initialState;
 
   let parsedAllocations = matchAllocationsWithAddresses(
```
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:79]: Btw, why do we need `parsedAllocations` and `cumulativeAllocation` as variables? Is it possible to use `const` instead?

File: test/unit/MerkleDrop.spec.js [hunk 2]
```diff
@@ -330,4 +483,29 @@ describe(">> MerkleDrop", () => {
       });
     });
   });
+
+  describe("# expireTranche", () => {
+    let proof, trancheIdx, expectedBalance, tree, parsedAllocations;
```
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:488]: Why are we defining `parsedAllocations` and `tree` if they aren't used below?

File: test/unit/MerkleDrop.spec.js [hunk 3]
```diff
@@ -247,10 +372,38 @@ describe(">> MerkleDrop", () => {
...
+      let proof, expectedBalance;
+
+      const initialState = {
+        ...commonState,
+        withProof: true,
+        zeroAllocation: true,
+      };
+      const nonExistentTrancheIdx = "5";
+
+      it("reverts 'Week cannot be in the future'", async () => {
```
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:386]: Just interesting, why exactly `Week cannot be in the future` for nonexistent tranche?

File: test/unit/MerkleDrop.spec.js [hunk 4]
```diff
@@ -219,9 +237,116 @@ describe(">> MerkleDrop", () => {
...
+        const expectedTotalBalance = utils
+          .parseEther(aliceSecondClaim)
+          .add(aliceFirstExpectedBalance);
+        expect(await v2TokenInstance.balanceOf(alice.address)).to.eq(
+          expectedTotalBalance
+        );
+      });
+
+      describe("$ with first tranche expired", () => {
+        it("lets alice claim her second claim", async () => {
```
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:327]: Is is correct that we cannot claim the first tranche after we claimed the second one? Or we won't be able to pick the first one only after it was expired by function call?

File: test/unit/MerkleDrop.spec.js (hunk 1 above)
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:79]:
```suggestion
  const parsedAllocations = matchAllocationsWithAddresses(
```

File: test/unit/MerkleDrop.spec.js (hunk 2 above)
Comment (BritikovKI) [inline test/unit/MerkleDrop.spec.js:488]:
```suggestion
    let proof, trancheIdx, expectedBalance;
```

File: test/unit/MerkleDrop.spec.js (hunk 1 above)
Comment (fabianschu) [reply test/unit/MerkleDrop.spec.js:79]: `parsedAllocations` can be made a constant. The other variable needs to remain overwritable for one test case. Thanks for spotting!

File: test/unit/MerkleDrop.spec.js (hunk 3 above)
Comment (fabianschu) [reply test/unit/MerkleDrop.spec.js:386]: Yea, I put off the brain and just copied what was already in the MerkleDrop here. I just changed the error message now. thanks

File: test/unit/MerkleDrop.spec.js (hunk 4 above)
Comment (fabianschu) [reply test/unit/MerkleDrop.spec.js:327]: As a user, we can claim basically from any tranche that
- is not expired
- has not yet been claimed by the same user

I added another test case to make this clear.

File: test/unit/MerkleDrop.spec.js (hunk 1 above)
Comment (BritikovKI) [reply test/unit/MerkleDrop.spec.js]: Roger, great

**Review by BritikovKI — APPROVED** [review]
Great work!

**Comment by Zitzak** [comment]
Some things I didn't spot. Nice to read through comments

**Review by Zitzak — APPROVED** [review]

**Pull request merged**